src/Documentation/grains/transactions.md

```diff
@@ -3,82 +3,110 @@ layout: page
-
-### Transactional State
-
-To support various transactional storage patterns the grain facing interface and supporting logic is pluggable.  We currently only support ITransactionalState.  This can be programmatically enabled on the silo host builder using UseTransactionalState.  To use a transactional state, a storage provider needs also be configured.
+To use transactions, the user needs to configure a data store.
+To support various data stores with transactions, the storage abstraction `ITransactionalStateStorage` has been introduced.
+This abstraction is specific to the needs of transactions, unlike generic grain storage (`IGrainStorage`).
+To use transaction-specific storage, the user can configure their silo using any implementation of `ITransactionalStateStorage`, such as Azure (`AddAzureTableTransactionalStateStorage`).
 
 Example:
 
 ``` csharp
 var builder = new SiloHostBuilder()
-    [...]
-    .AddAzureTableGrainStorageAsDefault(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”)
-    .UseInClusterTransactionManager()
-    .UseAzureTransactionLog(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”) 
-    .UseTransactionalState();
-
-var host = builder.Build();
-await host.StartAsync();
+    .AddAzureTableTransactionalStateStorage(“TransactionStore”, options =>
+    {
+        options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”);
+    })
+    .UseTransactions();
 ```
 
+For development purposes, if transaction specific storage is not available for the data store you need, an `IGrainStorage` implementation may be used instead.
+For any transactional state which does not have a store configured for it, transactions will attempt to fail over to grains storage using a bridge.
+Accessing transactional state via a bridge to grains storage will be less efficient and is not a pattern we intend to support long term, hence the recommendation that this only be used for development purposes.
+
 ## Programming Model
 
-For a grain to support transactions, transactional operations on a grain must be marked as being part of a transaction using the “Transaction” attribute.  Calls can be marked as “Required”, meaning the call can take part in a transaction among multiple other grains or as “RequiresNew” indicating that it will already start its own transaction.
+### Grain Interfaces
 
-Example:
+For a grain to support transactions, transactional operations on a grain interface must be marked as being part of a transaction using the “Transaction” attribute.
+The attribute needs indicate how the grain call behaves in a transactional environment.
+It indicates this via the transaction options below:
+
+- `TransactionOption.Create` - Call is transactional and will always create a new transaction context, even if called within an existing transaction context.
+- `TransactionOption.Join Call` - Logic is transactional but can only be called within the context of an existing transaction.
```

Comment:
`Join Call` -> `Join`

Reply:
- `TransactionOption.Join` - Call logic is transactional but can only be called within the context of an existing transaction.